packages/client/src/components/form/Button.tsx

```diff
@@ -0,0 +1,55 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
+ */
+import React from 'react'
+import { Button } from '@opencrvs/components/lib'
+import {
+  Ii18nButtonFormField,
+  IFormField,
+  IFormFieldValue
+} from '@client/forms'
+import { isFieldHttp } from '@client/forms/utils'
+
+interface ButtonProps {
+  fieldDefinition: Ii18nButtonFormField
+  children: React.ReactNode
+  fields: IFormField[]
+  setFieldValue: (name: string, value: IFormFieldValue) => void
+}
+
+export function ButtonField(props: ButtonProps) {
+  const getOnClickHandle = () => {
+    const trigger = props.fields.find(
+      (f) => f.name === props.fieldDefinition.options.trigger
+    )
+    if (trigger && isFieldHttp(trigger)) {
```

Comment:
Could we assume [via the form validations](https://github.com/opencrvs/opencrvs-core/blob/0a59d8b9b15a15f49f2e46b478999615ca9bf07d/packages/config/src/handlers/forms/validation.ts#L496) that a button always has a trigger and therefore we could assert it here?

```
const trigger = props.fields.find(...)!
```

I'm not sure if we do that granular field validation

Reply:
Sure, I think that would be great! Will work on it

Reply:
done https://github.com/opencrvs/opencrvs-core/pull/7574/commits/0f84e529b7be2b638fd342f0218623a6bfda7e13

![image](https://github.com/user-attachments/assets/ccc4a2dc-e481-497f-89ed-37d1ac29dcb0)